grove/remap/remap.go

```diff
@@ -497,7 +498,11 @@ func makeTo(tosJSON []RemapRuleToJSON, rule remapdata.RemapRule, baseTransport *
 				return nil, fmt.Errorf("error parsing to %v proxy_url: %v", to.URL, toJSON.ProxyURL)
 			}
 			to.ProxyURL = proxyURL
-			newTransport := *baseTransport
+			newTransport := http.Transport{}
+			const transportSize = unsafe.Sizeof(*baseTransport)
+			// Copy baseTransport to newTransport without failing the copylocks `go vet` check. See
+			// b7953cc239 for the reasoning behind copying the Transport.
+			copy((*[transportSize]byte)(unsafe.Pointer(&newTransport))[:], (*[transportSize]byte)(unsafe.Pointer(baseTransport))[:])
```

Comment:
I assume this is the only way to make `go vet` happy with the copy?

Reply:
I, too, was hoping there was a way to do this without `unsafe`

Reply:
We could do

```go
newTransport := *baseTransport.Clone()
```
but it would skip unexported fields like locks. We could use `reflect` to copy and set the unexported fields, but that would also involve using `unsafe`.

As for silencing the check on this line, Go offers no such option. From [go/issues#17058](https://github.com/golang/go/issues/17058#issuecomment-259012764):

> I am adamant that source code not be annotated to silence vet. That will reduce the options available but that is my position.

Reply:
We don't _need_ to copy the lock, we just need to copy the other fields we set from config, and we need the lock to not be the same as the other object. Default-initialization of the lock is fine.
 
We could also manually record and set all the transport fields we care about. I don't like that option, it's error-prone, it would silently fail if someone added a new option and forgot to add it to the "copy record."

I think `Clone` will work. All the fields we're setting in config are exported, and unless some function is added to Transport to set a private field, that shouldn't cause subtle bugs in the future.

Reply:
Using `Clone` in da61fe67d0.